### PR TITLE
Enable Rust on OSX ARM64

### PR DESCRIPTION
**Pull request by azat**
### Changelog category (leave one):
- Build/Testing/Packaging Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Enable Rust on OSX ARM64 (this will add fuzzy search in client with skim and prql language, though I don't think that are people who hosts ClickHouse on darwin, so it is mostly for fuzzy search in client I would say)

Right now it is disabled:

```
Jan 25 16:06:03 -- Adding contrib module corrosion (configuring with corrosion-cmake)
Jan 25 16:06:03 -- Rust is not available on aarch64-apple-darwin
Jan 25 16:06:03 -- Not using rust
```

I don't have idea why it had been disabled (back in https://github.com/ClickHouse/ClickHouse/pull/33435), but at least skim is compiled properly, so let's try

Follow-up for: #33435 (cc @BoloniniD)

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit 50f0a34ecb8d686d2b185351df444445ce4b8030 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/59272/50f0a34ecb8d686d2b185351df444445ce4b8030/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/59272/50f0a34ecb8d686d2b185351df444445ce4b8030/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

**Comment by azat** [comment]
I've tested locally and it had been successfully built, so should work, let's see what CI will tell.

**Review by alexey-milovidov — APPROVED** [review]

**Comment by azat** [comment]
>Performance Comparison [1/4] — Errors while building the report.
>Performance Comparison [2/4] — Errors while building the report.
>Performance Comparison [3/4] — Errors while building the report.
>Performance Comparison [4/4] — Errors while building the report.

- https://github.com/ClickHouse/ClickHouse/pull/59287

File: contrib/corrosion-cmake/CMakeLists.txt [hunk 1]
```diff
@@ -29,14 +28,14 @@ elseif(CMAKE_TOOLCHAIN_FILE MATCHES "linux/toolchain-aarch64")
     set(Rust_CARGO_TARGET "aarch64-unknown-linux-gnu")
 elseif((CMAKE_TOOLCHAIN_FILE MATCHES "darwin") AND (CMAKE_TOOLCHAIN_FILE MATCHES "x86_64"))
     set(Rust_CARGO_TARGET "x86_64-apple-darwin")
+elseif((CMAKE_TOOLCHAIN_FILE MATCHES "darwin") AND (CMAKE_TOOLCHAIN_FILE MATCHES "darwin"))
```
Comment (p1rattttt) [inline contrib/corrosion-cmake/CMakeLists.txt:31]: Кажется, что тут должно быть
elseif((CMAKE_TOOLCHAIN_FILE MATCHES "darwin") AND (CMAKE_TOOLCHAIN_FILE MATCHES "aarch64"))

Comment (azat) [reply contrib/corrosion-cmake/CMakeLists.txt:31]: What a dumb mistake, thank you! Can you please send a PR?

Comment (p1rattttt) [reply contrib/corrosion-cmake/CMakeLists.txt:31]: You are welcome! https://github.com/ClickHouse/ClickHouse/pull/59307

**Comment by nikitamikhaylov** [comment]
Hello, @azat! I believe this PR is the cause for the following linker warnings: 
```
ld: warning: object file (rust/prql/Debug/lib_ch_rust_prql.a[74](aarch_aapcs64.o)) was built for newer 'macOS' version (14.0) than being linked (11.0)
ld: warning: object file (rust/skim/Debug/lib_ch_rust_skim_rust.a[40](c0e230e4543ad627-lib.rs.o)) was built for newer 'macOS' version (14.0) than being linked (11.0)
ld: warning: object file (rust/skim/Debug/lib_ch_rust_skim_rust.a[46](6c3a6ae2099d4914-cxx.o)) was built for newer 'macOS' version (14.0) than being linked (11.0)
```
Which occur when you compile ClickHouse natively on Mac. Do you know how to fix it?

**Comment by azat** [comment]
Yep. This should be fixed by https://github.com/ClickHouse/ClickHouse/pull/59852

**Pull request merged**